distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Connection.enso

```diff
@@ -111,18 +102,43 @@ type Connection
             if all_fields then renamed else
                 renamed.select_columns (Column_Selector.By_Name ["Database", "Schema", "Name", "Type", "Description"])
 
-    ## ADVANCED
+    ## Set up a query returning a Table object, which can be used to work with data within the database or load it into memory.
 
-       Executes a raw query and returns the result as an in-memory Table.
+       Arguments:
+       - query: name of the table or sql statement to query.
+         If supplied as `Text`, the name is checked against the `tables` list to determine if a table or query.
+       - alias: optionally specify a friendly alias for the query.
+    query : Text | SQL_Query -> Text -> Database_Table
+    query self query alias="" = handle_sql_errors <| case query of
+        Text ->
+            self.query alias=alias <|
+                if self.tables.at 'Name' . to_vector . contains query then (SQL_Query.Table_Name query) else
+                    SQL_Query.Raw_SQL query
+        SQL_Query.Raw_SQL raw_sql ->
+            columns = self.jdbc_connection.fetch_columns raw_sql
+            name = if alias == "" then (UUID.randomUUID.to_text) else alias
+            ctx = IR.context_for_query raw_sql name
+            Database_Table.make_table self name columns ctx
+        SQL_Query.Table_Name name ->
+            ctx = IR.context_for_table name (if alias == "" then name else alias)
+            columns = self.jdbc_connection.fetch_columns (self.dialect.generate_sql (IR.Select_All ctx))
+            Database_Table.make_table self name columns ctx
+
+    ## Execute the query and return the results as a Table.
```

Comment:
I would make it more clear that this `Table` is loaded into memory. `query` also returns a `Table` but they do differ.